src/data/polynomial/coeff.lean

```diff
@@ -205,6 +205,24 @@ by simpa only [pow_one] using coeff_mul_X_pow p 1 n
 @[simp] theorem coeff_X_mul (p : R[X]) (n : ℕ) :
   coeff (X * p) (n + 1) = coeff p n := by rw [(commute_X p).eq, coeff_mul_X]
 
+theorem coeff_mul_monomial (p : R[X]) (n d : ℕ) (r : R) :
+  coeff (p * monomial n r) (d + n) = coeff p d * r :=
```

Comment:
This lemma and the following ones presumably could be proved also for `add_monoid_algebra`s, right?

Reply:
Maybe, mathlib does not have a definition of `coeff` for an `add_monoid_algebra`.

Reply:
I think we already have this lemma for `add_monoid_algebra`, as `mul_single_apply`. Either way, it's out of scope for this PR.